chat/src/hooks/useChatBotQueryData.ts

```diff
@@ -0,0 +1,35 @@
+import { useMemo } from "react";
+import { useChatState } from "./useChatState";
+import { useEditorContext, EditorContext } from "./useEditorContext";
+import { useUserDetails } from "./useUserDetails";
+import { v4 as uuidv4 } from "uuid";
+
+export type ChatBotQueryData = {
+  token: string;
+  username: string;
+  conversationId: string;
+  messageId: string;
+  editorContext?: EditorContext;
+};
+
+export function useChatBotQueryData(): ChatBotQueryData | null {
+  const userDetails = useUserDetails();
+  const { editorContext, isEditorContextChanged } = useEditorContext();
+  const { conversationMessages, currentConversation } = useChatState();
+  const messageId = useMemo(() => uuidv4(), []);
```

Comment:
`useMemo` used for optimization not for consistency. It's not promised to not run twice. In other words - move the `messageId` out of the component